addon/services/file-manager.js

```diff
@@ -1,43 +1,69 @@
 import Ember from 'ember';
 
+/**
+ * An Ember service for doing things to files.
+ * Essentially a wrapper for the Waterbutler API.
+ * http://waterbutler.readthedocs.io/
+ *
+ * @class file-manager
+ * @extends Ember.Service
+ */
+
 export default Ember.Service.extend({
     session: Ember.inject.service(),
     store: Ember.inject.service(),
 
-    // TODO: After performing a Waterbutler action, it takes a little while
-    // (maybe a minute, depending on caching options) for the OSF API to catch
-    // up and start serving the updated metadata.
-    //
-    // For actions on existing entities, this is fine, and file-manager
-    // uses info from WB to update the Ember store so everything looks right
-    // to the user, right away.
-    //
-    // For actions that create entities (addSubfolder, uploadFile, copy),
-    // this is a problem, because we need the OSF info (guid, WB links)
-    // before letting the user do anything to their new file. At the moment,
-    // after you create a file, you should go get a cup of coffee, come back,
-    // and hard reload.
-
-    ///////////////////// File-only actions /////////////////////
-
+    /**
+     * Download the contents of the given file.
+     *
+     * @method getContents
+     * @param {file} file A `file` model with `isFolder == false`.
+     * @return {Promise} Promise that resolves to the file contents or rejects
+     * with an error message.
+     */
     getContents(file) {
         let url = file.get('links').download;
         return this._waterbutlerRequest('GET', url);
     },
 
+    /**
+     * Upload a new version of an existing file.
+     *
+     * @method updateContents
+     * @param {file} file A `file` model with `isFolder == false`.
+     * @param {Object} contents A native `File` object or another appropriate
+     * payload for uploading.
+     * @return {Promise} Promise that resolves to the updated `file` model or
+     * rejects with an error message.
+     */
     updateContents(file, contents) {
```

Comment:
Yeah, that's a good idea, I could imagine that coming in handy for adding functionality without messing with the interface or adding state. I'll do that.